File: libs/language/src/lib/mapping/PrimitiveConverter.ts
import {
  ZodBoolean,
  ZodBigInt,
  ZodNumber,
  ZodString,
  ZodEnum,
  ZodNativeEnum,
} from 'zod';
import { BlueNode } from '../model';
import { Converter } from './Converter';
import { ValueConverter } from './ValueConverter';

export class PrimitiveConverter implements Converter {
  convert(
    node: BlueNode,
    targetType:
      | ZodString
      | ZodNumber
      | ZodBoolean
      | ZodBigInt
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      | ZodEnum<any>
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      | ZodNativeEnum<any>,
  ) {
    return ValueConverter.convertValue(node, targetType);
  }
}

File: references/java-sdk/src/main/java/blue/language/mapping/PrimitiveConverter.java
package blue.language.mapping;

import blue.language.model.Node;

import java.lang.reflect.Type;

class PrimitiveConverter implements Converter<Object> {
    @Override
    public Object convert(Node node, Type targetType) {
        if (targetType instanceof Class<?>) {
            return ValueConverter.convertValue(node, (Class<?>) targetType);
        } else {
            throw new IllegalArgumentException("Unsupported target type for primitive conversion: " + targetType);
        }
    }
}